staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -242,10 +248,128 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 		_, err = w.Write(data)
 		return err
 	}
+	if s.options.StreamingCollectionsEncoding {
+		typeMeta, listMeta, items, err := meta.GetListMeta(obj)
+		if err == nil {
+			return streamingEncodeList(w, typeMeta, listMeta, items)
+		}
+		list, ok := obj.(*unstructured.UnstructuredList)
+		if ok {
+			return streamingEncodeUnstructured(w, list)
+		}
+	}
 	encoder := json.NewEncoder(w)
 	return encoder.Encode(obj)
 }
 
+func streamingEncodeUnstructured(w io.Writer, list *unstructured.UnstructuredList) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	for k, v := range list.Object {
+		err := encodeValue(w, nil, k, []byte(":"))
```

Comment:
unstructured lists can include `items` in Object... won't this double-encode them?

https://github.com/kubernetes/kubernetes/blob/c3f3fdc1aa62002a58bec1141fe69e86bbb27491/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_list.go#L115-L116

Reply:
Right missed that, will add a test for that. 

https://github.com/kubernetes/kubernetes/blob/f1b3fdf7e6d40714b1a43757221832aa1c4a49d1/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L409-L418

Reply:
Done